cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -74,9 +87,29 @@ public void initialize(WorkflowContext context) throws Exception {
              programType != null ? programType.name() : null, programName);
   }
 
+  public void setTxClient(TransactionSystemClient txClient) {
```

Comment:
It is set in the constructor, but the problem is that in WorkflowDriver's initialize, we use reflection to get back a new instance of the class, so we need to set the txClient again explicitly